Automattic/jetpack
# Fix build for plugins/boost when using TypeScript inside js-packages

This PR fixes the failing build for `plugins/boost` when using TypeScript inside `js-packages`. Please see the related issue for details

Fixes #23961 

#### Changes proposed in this Pull Request:

* Adds a custom tsconfig.json file for rollup named `rollup-tsconfig.json`.
* Passes the new `rollup-tsconfig.json` path to `@rollup/plugin-typescript` in `rollup.config.js`.
* Converts `numberFormat` in `js-packages/components` to TypeScript as a proof for the fix

#### Jetpack product discussion
https://github.com/Automattic/jetpack/pull/23889#discussion_r850436961

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* Checkout the PR branch
* Run `jetpack build plugins/boost `
* Confirm that build is successful


#### Additional/Optional steps

* Convert any other component inside `js-packages/components` to TypeScript
* Run `jetpack build plugins/boost `
* Confirm that build is successful
